src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -1436,5 +1438,91 @@ protected static bool FilterException(Exception e)
             // Nothing fancy, so don't catch
             return false;
         }
+
+        #region FolderNames
```

Comment:
I guess it's fine for the legacy contained language scenario, since we're there's still so much other legacy going on that this isn't really the worst bit? :smile: